validation-core/src/main/scala/tagless/Solver.scala

```diff
@@ -0,0 +1,56 @@
+package jto.validation
+package v3.tagless
+
+import shapeless.{ ::, HNil, HList, Generic }
+import cats.Functor
+
+// TODO: Add helpful implicit not found message
+trait Solver[I, O] {
```

Comment:
yeah this is more a POC than anything else. Not sure it'll make it in the final version.